site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -377,18 +401,41 @@ public function ajaxUploadCourseMaterialsFiles() {
 
                         foreach ($zfiles as $zfile) {
                             $path = FileUtils::joinPaths( $upload_path, $zfile );
-                            $json[$path] = [
-                                'checked' => '1',
-                                'release_datetime' => $release_time
-                            ];
+                            if(!(is_null($sections))){
+                                $sections = @explode(",", $sections);
+                                if($sections == null){
+                                    $sections = [];
```

Comment:
If `$sections` is null here what is the point of adding an empty array to the json file, I think it would make sense in
the else condition to just not put anything

Reply:
If the json file contains an empty array as sections, the user selected to display only certain sections than didn't choose any sections. If the json file contains nothing as sections, then the user selected to display to all sections. These are two distinct cases that have to be treated differently.

Reply:
resolving based on meeting - didn't know this was expected behavior